h2/src/main/org/h2/command/ddl/AlterTableAlterColumn.java

```diff
@@ -156,9 +159,13 @@ public int update() {
                 } else if (!oldColumn.isNullable() && newColumn.isNullable()) {
                     checkNullable(table);
                 }
+                if (oldColumn.getVisible() ^ newColumn.getVisible()) {
+                    oldColumn.setVisible(newColumn.getVisible());
+                }
                 convertAutoIncrementColumn(table, newColumn);
                 copyData(table);
             }
+            table.setModified();
```

Comment:
The select query seems to be retrieved from cache unless I call this `table.setModified()`. The cached query has outdated column expression list (wildcard already expanded).